src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -684,17 +684,19 @@ private static StripeResponse getMultipartStripeResponse(
   }
 
   private static void handleApiError(String responseBody, int responseCode, String requestId)
-      throws InvalidRequestException, AuthenticationException,
-      CardException, ApiException {
+      throws ApiException, AuthenticationException, CardException, IdempotencyException,
+      InvalidRequestException {
     LiveStripeResponseGetter.StripeError error = ApiResource.GSON.fromJson(responseBody,
         LiveStripeResponseGetter.StripeErrorContainer.class).error;
     switch (responseCode) {
       case 400:
-        throw new InvalidRequestException(error.message, error.param, requestId, error.code,
-            responseCode, null);
       case 404:
-        throw new InvalidRequestException(error.message, error.param, requestId, error.code,
-            responseCode, null);
+        if (error.type.equals("idempotency_error")) {
```

Comment:
Why is this in `case 404`?

Reply:
`switch` in Java is *fall through*, i.e. all `case` blocks are executed until a `break` statement appears. So this code applies to both `case 400:` and `case 404:`, i.e. exactly what stripe-ruby does with `when 400, 404`.

Server-side, I think that `idempotency_error`s will always have a 400 status code. We should revamp the error handling logic in all libraries to prefer using the error's `type` over the status code IMHO, but that's out of scope here.

Reply:
I've never been a fan for using this case hack tbh, especially for something as key as this. It's really easy to miss when changing in the future. But up to you in that case, feel free to self-approve/merge

Reply:
I wouldn't call this a hack, it's a well documented behavior (https://docs.oracle.com/javase/tutorial/java/nutsandbolts/switch.html), though I agree it's not the most readable syntax ever. In this case it lets up slightly simplify the logic (since `invalid_request_error` can be either a 400 or a 404).

Going to merge this one, and will try to revamp the error handling logic later.